### ISSUE TITLE
v1.0-beta requires nibrary older than 03/Aug/2024 commit 01ec9a0

### ISSUE THREAD

**Issue by byebrid** [dmritrekker/trekker#20]
# Summary
To build trekker v1.0-beta from source, the README says to first [install nibrary](https://github.com/nibrary/nibrary). **However, the latest commit of nibrary that I could successfully build trekker with was [nibrary commit 01ec9a0](https://github.com/nibrary/nibrary/commit/01ec9a0d03918fa7a3c304d3352338c7420ed5ec) from 02/Aug/2024.**

# Context
- OS: Rocky 9
- Building from source (with system zlib and eigen for nibrary).
- Version of trekker: [1.0-beta](https://github.com/dmritrekker/trekker/releases/tag/v1.0-beta).

# Problem in detail
First I had tried building trekker against [nibrary commit f47dc50](https://github.com/nibrary/nibrary/commit/f47dc50481d37fbb2bbc65af73acb580185c3f93), but I got this error:
```
/apps/src/TREKKER/1.0-beta/trekker/src/cmd/filter.cpp: In function ‘void run_filter()’:
/apps/src/TREKKER/1.0-beta/trekker/src/cmd/filter.cpp:102:36: error: ‘class NIBR::Pathway’ has no member named ‘noEdgeSeed’
  102 |     if((!seedList.empty()) && (!pw.noEdgeSeed(!allowEdgeSeeds))) return;
...
```

That is because [nibrary commit b918e65](https://github.com/nibrary/nibrary/commit/b918e65442406982a80741471243664d8d952ac0) removed the `noEdgeSeed` member.

I thus tried the parent of that nibrary commit instead, which is [nibrary commit 01ec9a0](https://github.com/nibrary/nibrary/commit/01ec9a0d03918fa7a3c304d3352338c7420ed5ec). However, building trekker against this then failed at the linking stage:
```
/usr/bin/ld: CMakeFiles/trekker.dir/src/cmd/trekker.cpp.o: in function `run_trekker()':
trekker.cpp:(.text+0x107b): undefined reference to `NIBR::Trekker::seed_surface_useInside(bool)'
collect2: error: ld returned 1 exit status
```

This is because `seed_surface_useInside` had been removed in [nibrary commit 01ec9a0](https://github.com/nibrary/nibrary/commit/01ec9a0d03918fa7a3c304d3352338c7420ed5ec) from 03/Aug/2024.

# Solution
I then tried the parent of this commit, [nibrary commit 7e86eda](https://github.com/nibrary/nibrary/commit/7e86eda8b0894a6d561b37ea71aab4cd0cd693a8), which actually built successfully.

**Should this be documented in the trekker and/or nibrary READMEs?** I see that trekker and nibrary are being developed in parallel so I'm not sure if you intend to resolve this inconsistency soon, but it's certainly not obvious right now how to get a successful build going.

**Comment by baranaydogan** [comment]
Thank you very much for pointing this out. I now fixed this problem by updating trekker master branch, so it works with nibrary master. Our intention is to keep nibrary and trekker sync'ed. And indeed, these should be written in the documentation. One thing we are working on is to create separate dev and test branches for trekker and nibrary, we are also switching to a better versioning system. With these we want to create matching tags between nibrary and trekker. These are still ongoing efforts, and hopefully they will be out soon. But to fix today's immediate problem, I updated trekker's master branch, which compiles with nibrary master.
